src/siteSettings/views/index.tsx

```diff
@@ -35,167 +32,149 @@ export const SiteSettings: React.FC<SiteSettingsProps> = ({ params }) => {
   const notify = useNotifier();
   const intl = useIntl();
 
-  const handleAddKeySuccess = (data: AuthorizationKeyAdd) => {
-    if (data.authorizationKeyAdd.errors.length === 0) {
-      notify({
-        status: "success",
-        text: intl.formatMessage(commonMessages.savedChanges)
-      });
-      navigate(siteSettingsUrl());
+  const {
+    data: siteSettingsQueryData,
+    loading: siteSettingsQueryLoading
+  } = useSiteSettingsQuery({
+    displayLoader: true,
+    variables: {}
+  });
+
+  const [authorizationKeyAdd, authorizationKeyAddOpts] = useAuthorizationKeyAdd(
+    {
+      onCompleted: data => {
+        if (data.authorizationKeyAdd.errors.length === 0) {
+          notify({
```

Comment:
This chunk could be moved to a function since it happens 3 times here 🚔

```
const handleComplete = (errors) => {
 if (errors?.length === 0) {
     notify({
        status: "success",
        text: intl.formatMessage(commonMessages.savedChanges)
     });
     navigate(siteSettingsUrl());
   }
}

Reply:
Dont want to mess with this pattern in this PR.  Each notification should have a different message by the way, but it's out of the scope.